pkg/backend/openshiftcluster/create.go

```diff
@@ -37,17 +38,26 @@ func (m *Manager) Create(ctx context.Context) error {
 	var err error
 
 	resourceGroup := m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID[strings.LastIndexByte(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')+1:]
+	var sshKey *rsa.PrivateKey
+	if m.doc.OpenShiftCluster.Properties.SSHKey == nil {
+		sshKey, err = rsa.GenerateKey(rand.Reader, 2048)
+		if err != nil {
+			return err
+		}
+		m.doc.OpenShiftCluster.Properties.SSHKey = x509.MarshalPKCS1PrivateKey(sshKey)
+		if err != nil {
+			return err
+		}
+	} else {
```

Comment:
sorry, my last review was probably erroneous, I now see that this needs to go in PatchWithLease and that the ParsePKCS1PrivateKey is necessary and belongs below.  Please go back to what you had before.

Reply:
:(